zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -53,48 +81,58 @@ impl<F: Field> ExecutionGadget<F> for PushGadget<F> {
         //                           ▼                     ▼
         //   [byte31,     ...,     byte2,     byte1,     byte0]
         //
-        for idx in 0..32 {
+        // First is_pushed (resp. is_padding) will always be 1 (resp. 1)
+        let mut pu_prev = 1.expr();
+        let mut pa_prev = 1.expr();
+        for (idx, (pu, pa)) in is_pushed.iter().zip(is_padding.iter()).enumerate() {
             let byte = &value.limbs[idx];
             let index = cb.curr.state.program_counter.expr() + opcode.expr()
```

Comment:
Suggestion: we have
```rust
let num_pushed = opcode.expr() - OpcodeId::PUSH1.as_u64().expr() + 1.expr();
```
So this would be simplified like:
```rust
let index = cb.curr.state.program_counter.expr() + num_pushed - idx.expr();
```